src/composables/useCurrentTime.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { createSharedComposable } from '@vueuse/core'
+import { onUnmounted, readonly, ref } from 'vue'
+import type { DeepReadonly, Ref } from 'vue'
+
+/**
+ * Composable to get current time (as Date object)
+ * @param [precision] - precision in milliseconds (defaults to one minute)
+ * @return Date reactive object with current time
+ */
+export function useCurrentTimeComposable(precision: number = 60_000): DeepReadonly<Ref<Date>> {
+	let timeout: NodeJS.Timeout = null
```

Comment:
`null` is as `object`

For this variable we can just keep it undefined

```suggestion
	let timeout: NodeJS.Timeout
```

Reply:
Also `NodeJS.Timeout` is not a correct type for a web-browser. But probably should be fixed on tsconfig level.